### PR TITLE
add setting to configure when to raise the user unload event and remove user always before requesting sign-out

### LINKED ISSUE #1379: UserManager signoutRedirect not working
To logout I use the[ signoutRedirect method of the UserManager](https://authts.github.io/oidc-client-ts/classes/UserManager.html#signoutRedirect).
That worked perfectly fine with version 2.4.0.

With version 3.0.0 the user is no longer properly logged out.

oidc-client-ts correctly redirects the user agent to the auth server with id_token_hint and post_logout_redirect_uri.
The auth server performs the logout as usual and redirects the user agent back to the application.

The difference compared to version 2.4.0 is, that the UserManager still holds a user object, accesible by the [getUser method](https://authts.github.io/oidc-client-ts/classes/UserManager.html#getUser).

Causing the issue should be the removal of [this line in the _signoutStart method](https://github.com/authts/oidc-client-ts/blob/v2.4.0/src/UserManager.ts#L552).

Is this a bug or should I change some things in my application?

### PR DESCRIPTION
**Pull request by pamapa**
Closes/fixes #1379

### Checklist

- [ ] This PR makes changes to the public API 
- [x] I have included links for closing relevant issue numbers

### CODE REVIEW

**Comment by pamapa** [comment]
@PSanetra Can you review and test?

- unload event is
  - deferred to callback (default, `raiserUserUnloadEventBeforeSignoutRequest=false`)
  - done before requesting the sign-out  (option: `raiserUserUnloadEventBeforeSignoutRequest=true`)
- remove user from storage is always (independent of option) done before requesting the sign-out

Any better name for `raiserUserUnloadEventBeforeSignoutRequest`?

File: src/UserManager.ts [hunk 1]
```diff
@@ -641,8 +654,9 @@ export class UserManager {
         const signoutResponse = await this._client.processSignoutResponse(url);
         logger.debug("got signout response");
 
-        await this.removeUser();
-        logger.debug("user removed");
+        if (!this.settings.raiserUserUnloadEventBeforeSignoutRequest) {
```
Comment (pamapa) [inline src/UserManager.ts:657]: maybe we should just call `await this._removeUser(true)` here, it should not hurt and is simpler (less code)

Comment (PSanetra) [reply src/UserManager.ts:657]: I think this should be ok as a signed in user would be unexpected in any case, right?

**Comment by PSanetra** [comment]
I think `raiserUserUnloadEventBeforeSignoutRequest` might be renamed to `noSignoutCallbacksSupported` or `signoutCallbackSupported` with `true` as default if the value is `undefined`.

The user/ developer is expected to set `signoutCallbackSupported` to false if he expects never to call any signout callback.

**Comment by pamapa** [comment]
I think this solution is too complicated, i do not like an option for this at all. The previous behavior (<v3.0) was dead simple

**Pull request closed without merging**